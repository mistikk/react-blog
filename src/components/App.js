import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Header from './Header';
import Footer from './Footer';
import Content from './Content';

import Home from '../routes/Home';
import Start from '../routes/Start';

const App = () =>
  <div>
    <Header />
    <Content>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/basla" component={Start} />
      </Switch>
    </Content>
    <Footer />
  </div>;

export default App;
