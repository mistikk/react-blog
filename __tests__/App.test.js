import React from 'react';
import { shallow } from 'enzyme';

import App from '../src/App';

it('renders correctly', () => {
  const wrapper = shallow(<App />);

  expect(wrapper).toMatchSnapshot();
});
