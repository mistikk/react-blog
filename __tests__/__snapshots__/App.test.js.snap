exports[`test renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App">
    <div
        className="App-header">
        <img
            alt="logo"
            className="App-logo"
            src="logo.svg" />
        <h2>
            Welcome to React
        </h2>
    </div>
    <p
        className="App-intro">
        To get started, edit 
        <code>
            src/App.js
        </code>
         and save to reload.
    </p>
</div>,
  "nodes": Array [
    <div
      className="App">
      <div
            className="App-header">
            <img
                  alt="logo"
                  className="App-logo"
                  src="logo.svg" />
            <h2>
                  Welcome to React
            </h2>
      </div>
      <p
            className="App-intro">
            To get started, edit 
            <code>
                  src/App.js
            </code>
             and save to reload.
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App">
          <div
                    className="App-header">
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg" />
                    <h2>
                              Welcome to React
                    </h2>
          </div>
          <p
                    className="App-intro">
                    To get started, edit 
                    <code>
                              src/App.js
                    </code>
                     and save to reload.
          </p>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="App">
          <div
                    className="App-header">
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg" />
                    <h2>
                              Welcome to React
                    </h2>
          </div>
          <p
                    className="App-intro">
                    To get started, edit 
                    <code>
                              src/App.js
                    </code>
                     and save to reload.
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
